eslint-plugin-expensify/no-negated-variables.js

```diff
@@ -2,30 +2,62 @@ const _ = require('underscore');
 const lodashGet = require('lodash/get');
 const message = require('./CONST').MESSAGE.NO_NEGATED_VARIABLES;
 
+const BANNED_SUBSTRINGS = [
+    'not',
+    'isnot',
+    'cannot',
+    'shouldnot',
+    'cant',
+    'dont',
+];
+
+const NOTABLE_EXCEPTIONS = [
+    'notification',
+    'notch',
+    'note',
+    'notable',
+    'notion',
+    'notice',
+];
+
+// Declared up front to suppress eslint no-use-before-define
+// Mutually recursive functions are a valid exception to this rule. See https://github.com/eslint/eslint/issues/12473
+let isNegatedVariableName;
```

Comment:
NAB, probably better to just suppress this rule than to declare/explain here. What does "mutually recursive" mean?

Reply:
It just means two functions recursively call eachother:

1. `isNegatedVariableName` name calls `isFalsePositive`
1. `isFalsePositive` calls `isNegatedVariableName`,
1. etc..
1. etc..

Then there's no way to declare both functions before using them unless you create a reference like this (or just suppress the rule). The only reason I went with approach instead of just suppressing the rule inline was so I could include a comment to explain why suppressing the rule is okay.

Reply:
Ok, I think we'd normally suppress the rule before doing what you are doing here, but it's a NAB. If you agree that the rule could be suppressed and this would be more consistent with how we handle this situation elsewhere then fix it up.